docs/source/jobs.rst

```diff
@@ -109,11 +109,83 @@ Job-specific optional keys:
 - ``name``: Human-readable name/label of the job
 - ``filter``: :ref:`filters` (if any) to apply to the output
 - ``max_tries``: Number of times to retry fetching the resource
 - ``diff_tool``: Command to a custom tool for generating diff text
 - ``compared_versions``: Number of versions to compare for similarity
 - ``kind`` (redundant): Either ``url``, ``shell`` or ``browser``.  Automatically derived from the unique key (``url``, ``command`` or ``navigate``) of the job type
 
+
+.. _comparison_filter:
+
+``comparison_filter``
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``comparison_filter`` filters the output of the unified diff to keep only addition or deleted lines
+ - A value of `additions` will cause reports to contain only lines that are added by the diff (no deletions).
+ - A value of `deleted` key will cause reports to contain only lines that are deleted by the diff (no additions).
+
+`comparison_filter: additions` is extremely useful for monitoring new content on sites where content gets added while old content "scrolls" away.
+
+Because lines that are modified generate both a deleted and an added line by the diff, this filters always displays modified lines.
+
+As a safeguard, `additions` will warn when 75% of more of the change consists of deletions.
+
+
+Sample output for `additions`:
+
+.. code-block:: none
+
+   ---------------------------------------------------------------------------
+   CHANGED: https://example.com
+   ---------------------------------------------------------------------------
+   ... @   Sat, 23 May 2020 00:00:00 +0000
+   +++ @   Sat, 23 May 2020 01:00:00 +0000
+   -**Comparison type: Additions only**
+   @@ -1,2 +1,2 @@
+   +This is a line that has been added or changed
+
+Sample output for `deletions`:
+
+.. code-block:: none
+
+   ---------------------------------------------------------------------------
+   CHANGED: https://example.com
+   ---------------------------------------------------------------------------
+   --- @   Sat, 23 May 2020 00:00:00 +0000
+   ... @   Sat, 23 May 2020 01:00:00 +0000
+   +**Comparison type: Deletions only**
+   @@ -1,2 +1,2 @@
+   -This is a line that has been deleted or changed
+
+Sample output for `additions` when the 75% deletions safeguard is triggered:
+
+.. code-block:: none
+
+   ---------------------------------------------------------------------------
+   CHANGED: https://example.com
+   ---------------------------------------------------------------------------
+   ... @   Sat, 23 May 2020 00:00:00 +0000
+   +++ @   Sat, 23 May 2020 01:00:00 +0000
+   -**Comparison type: Additions only**
+   .** No additions (only deletions)
+   --- WARNING: 39 lines deleted; suggest checking source
+   ---------------------------------------------------------------------------
+
+
+Workaround: due to legacy logic in urlwatch, we are unable to suppress reporting when all changes are filtered out, so a message is added instead:
```

Comment:
It would be great if we could just fix the legacy logic so that instead of documenting the workaround we'll fix it properly. A quick look at `reporters.py` shows that each job state gets added to the summary and details like this:

```
        summary = []                                                                                                     
        details = []                                                                                                     
        for job_state in self.report.get_filtered_job_states(self.job_states):                                           
            summary_part, details_part = self._format_output(job_state)                                                  
            summary.extend(summary_part)                                                                                 
            details.extend(details_part)                                                                                 
```

And `self._format_output()` there calls `self._format_content()` which in turn calls `self.unified_diff()`, so it would be possible that if `self.unified_diff()` returns an empty string (or `None`), then `_format_output()` handles that and the return value of that function is handled on the caller site. The only tricky thing is that the "minimal" reporter doesn't call the diffing at all, so with the minimal reporter, changes (as summary line) would always be displayed, but that's likely fine, as it doesn't affect the diff output then, anyway.

Reply:
Thanks for the pointers.  Will take a look at it.  I am still overly confused at the Russian dolls here... :)

Reply:
Thanks for the refactoring, I got it to work!